pkg/reconciler/v1alpha1/autoscaling/hpa/resources/hpa.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"math"
+
+	"github.com/knative/serving/pkg/apis/autoscaling/v1alpha1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func MakeHPA(pa *v1alpha1.PodAutoscaler) *autoscalingv1.HorizontalPodAutoscaler {
+	min, max := pa.ScaleBounds()
+	if max == 0 {
+		max = math.MaxInt32 // default to no limit
+	}
+	hpa := &autoscalingv1.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pa.Name,
+			Namespace:       pa.Namespace,
+			Labels:          pa.Labels,
+			Annotations:     pa.Annotations,
+			OwnerReferences: pa.OwnerReferences,
```

Comment:
This should be a new OwnerReference to the `pa` not a mirror of the `pa`'s OwnerReferences.